Commands/Public/archive.js

```diff
@@ -24,39 +31,49 @@ module.exports = async ({ Constants: { Colors } }, documents, msg, commandData)
 				},
 			});
 		}
-		messages.forEach(message => {
-			archive.push({
-				author: {
-					username: message.author.username,
-					discriminator: message.author.discriminator,
-					id: message.author.id,
-					bot: message.author.bot,
-					avatarURL: message.author.displayAvatarURL(),
-				},
-				createdAt: `${moment(message.createdAt).tz("Europe/London").format(`DD[.]MM[.]YYYY [at] HH[:]mm[:]ss [UTC]Z`)}`,
-				content: message.content,
-				cleanContent: message.cleanContent,
-				editedAt: `${message.editedAt ? moment(message.editedAt).tz("Europe/London").format(`DD[.]MM[.]YYYY [at] HH[:]mm[:]ss [UTC]Z`) : "Message wasn't edited"}`,
-				embeds: message.embeds.map(e => ({
-					author: e.author || {},
-					color: e.hexColor ? `0x${e.hexColor.replace("#", "").toUpperCase()}` : null,
-					description: e.description,
-					fields: e.fields || [],
-					footer: e.footer || {},
-					image: e.image ? e.image.url : null,
-					timestamp: `${e.timestamp}`,
-					title: e.title,
-					type: e.type,
-					url: e.url || null,
-				})),
-				id: message.id,
-			});
-		});
-		const fileBuffer = Buffer.from(JSON.stringify(archive, null, 2));
+		messages = messages.map(message => ({
+			author: {
+				username: message.author.username,
+				discriminator: message.author.discriminator,
+				id: message.author.id,
+				bot: message.author.bot,
+				avatarURL: message.author.displayAvatarURL(),
+			},
+			id: message.id,
+			createdAt: moment(message.createdAt).tz("Europe/London").format(`DD[.]MM[.]YYYY [at] HH[:]mm[:]ss [UTC]Z`),
+			content: message.content || "",
+			cleanContent: message.cleanContent || "",
+			editedAt: message.editedAt ? moment(message.editedAt).tz("Europe/London").format(`DD[.]MM[.]YYYY [at] HH[:]mm[:]ss [UTC]Z`) : null,
+			embeds: message.embeds.map(e => ({
+				author: e.author ? {
+					name: e.author.name || null,
+					url: e.author.url || null,
+					iconURL: e.author.iconURL || null,
+				} : {},
+				color: e.hexColor ? `0x${e.hexColor.replace("#", "").toUpperCase()}` : null,
+				url: e.url || null,
+				description: e.description || "",
+				fields: e.fields || [],
+				footer: e.footer ? {
+					text: e.footer.text || null,
+					iconURL: e.footer.iconURL || null,
+				} : {},
+				image: e.image ? e.image.url : null,
+				thumbnail: e.thumbnail ? e.thumbnail.url : null,
+				timestamp: e.timestamp || null,
+				title: e.title || null,
+				type: e.type || null,
+			})),
+			attachments: message.attachments.map(a => ({
+				name: a.name,
+				url: a.attachment,
+			})),
+		}));
+		const fileBuffer = Buffer.from(JSON.stringify(messages, null, 2));
 		try {
 			await msg.channel.send({
-				content: `Here you go! ✅`,
-				files: [new Attachment(fileBuffer, `archive-${msg.guild.name}-${msg.channel.name}-${Date.now()}.json`)],
+				content: `Here you have the last ${count} messages! ✅`,
+				files: [new Attachment(fileBuffer, `archive-${msg.guild.name}-${msg.channel.name}-${count}-${Date.now()}.json`)],
```

Comment:
I kinda disagree with it being in the file name, but I'd agree with it being in the actual JSON. Something like:
```json
{
  "totalMessages": messages.size,
  "messages": messages.map()
}
```

Reply:
While we're at it, I could also add other metadata in the file such as archive creation time, server name etc.

Reply:
Sure